### DISCUSSION TITLE
Discussing the `Output` command

### DISCUSSION THREAD

**Discussion by mxcl** [charmbracelet/vhs#177] [General]
For me specifying output location in the file is a little “off”. But I get that if you didn't have this people would ask for it.

However I do feel we should be able to omit it (maybe that already works), and more importantly, specify this on the command line (where it would act as an override).

The reason I bring this up is I tried to execute `vhs` for https://github.com/charmbracelet/vhs/blob/main/examples/demo.tape and it fails because the `Output` was specified in the file and is set to `examples/demo.gif` and the examples directory did not exist for me (I downloaded the `.tape` file).

Now, should `vhs` create the examples directory? Perhaps. My main point though is I was surprised that a command line tool would have the output hardcoded in its input file and surprised that the default behavior resulted in no output (because it failed to write).

I was expecting `vhs foo.tape` to output `./foo.gif` as its default I guess.

Anyway, discussion opened.

**Comment by maaslalani** [comment]
This is a great point, `Output` should definitely have the option to be overridden on the command-line.

The reason why it is nice to have the output specified in the tape file also is for actions that regenerate all tape files in a repo. i.e.

```
vhs **/*.tape
```

However, I totally agree this should be overridable through a CLI flag:

```
vhs demo.tape --output examples/demo.gif
```

**Reply by maaslalani** [reply]
Also, for what it is worth, the `Output` is optional and will make a GIF in the `out.gif` location by default.

**Comment by maaslalani** [comment]
Btw, this should be available shortly 🙂 

* https://github.com/charmbracelet/vhs/pull/178

```bash
vhs demo.tape -o demo.gif -o demo.webm
```

**Comment by mxcl** [comment]
FYI `tea` now supports VHS so this will work once output can be overridden:

```
$ sh <(curl tea.xyz) https://github.com/charmbracelet/vhs/blob/main/examples/demo.tape --output demo.gif
tea: downloading script
tea: installed: ~/.tea/libevent.org/v2.1.12
tea: installed: ~/.tea/libuv.org/v1.44.2
tea: installed: ~/.tea/github.com/json-c/json-c/v0.16.0
tea: installed: ~/.tea/libwebsockets.org/v4.3.2
tea: installed: ~/.tea/ffmpeg.org/v5.1.2
tea: installed: ~/.tea/tsl0922.github.io/ttyd/v1.7.2
tea: installed: ~/.tea/charm.sh/vhs/v0.2.0

Output .gif examples/demo.gif
Require echo
Set FontSize 32
Set Width 1200
Set Height 600
Set LoopOffset 50%
Type echo 'Welcome to VHS!'
Sleep 500ms
Enter 1
Sleep 5s
Creating GIF...
ffmpeg version 5.1.2 Copyright (c) 2000-2022 the FFmpeg developers
  built with Apple clang version 14.0.0 (clang-1400.0.29.102)
  configuration: --prefix=/opt/ffmpeg.org/v5.1.2
  libavutil      57. 28.100 / 57. 28.100
  libavcodec     59. 37.100 / 59. 37.100
  libavformat    59. 27.100 / 59. 27.100
  libavdevice    59.  7.100 / 59.  7.100
  libavfilter     8. 44.100 /  8. 44.100
  libswscale      6.  7.100 /  6.  7.100
  libswresample   4.  7.100 /  4.  7.100
Input #0, image2, from '/var/folders/b6/td46wpyx6r9fw4mc2tfwybw40000gn/T/vhs4282268736/frame-text-%05d.png':
  Duration: 00:00:11.92, start: 0.000000, bitrate: N/A
  Stream #0:0: Video: png, rgba(pc), 2052x888, 25 fps, 25 tbr, 25 tbn
Input #1, image2, from '/var/folders/b6/td46wpyx6r9fw4mc2tfwybw40000gn/T/vhs4282268736/frame-cursor-%05d.png':
  Duration: 00:00:11.92, start: 0.000000, bitrate: N/A
  Stream #1:0: Video: png, rgba(pc), 2052x888, 25 fps, 25 tbr, 25 tbn
examples/demo.gif: No such file or directory
```
